app/js/controllers/actions/event/event-list-ctrl.js

```diff
@@ -0,0 +1,299 @@
+	Restangular.all('people')
+		.getList()
+		.then(function(person) {
+			_.each(person, function(element) {
+				personIdToName[element.id] = element.attributes.name;
+			});
+		})
+		.then(function(){
+			Restangular.all(resourceName)
+			.getList()
+			.then(function(data) {
+				$scope.data = data;
+				if (resourceId) {
+					var index = _.findIndex($scope.data, {id: resourceId});
+					$scope.model = $scope.data[index];
+				}
+
+				_.each($scope.data, function(element) {
+					//mapping eventID to venue names with links
+					if (element.relationships.venue.data != null) {
+						// !! change venueURL to the venue page once the venue override page is ready
+						var venueURL = "https://api.tnyu.org/v3/venues/" + element.relationships.venue.data.id;
+						$scope.eventDetails.venue[element.id] = "<a href=" + venueURL + ">" + venuesIdToName[element.relationships.venue.data.id] + "</a>";
+					};//end
+
+					//mapping eventID to categories
+					if (element.attributes.categories.length > 0) {
+						element.attributes.categories.forEach(category => {
+							if ($scope.eventDetails.categories[element.id] == undefined) {
+								$scope.eventDetails.categories[element.id] = category;
+							} else {
+								var string = $scope.eventDetails.categories[element.id];
+								string = string + ", " + category;
+								$scope.eventDetails.categories[element.id] = string;
+							}
+						})
+					};//end
+
+					//mapping eventID to aims
+					if (element.attributes.aims.length > 0) {
+						element.attributes.aims.forEach(aim => {
+							if ($scope.eventDetails.aims[element.id] == undefined) {
+								$scope.eventDetails.aims[element.id] = aim;
+							} else {
+								var string = $scope.eventDetails.aims[element.id];
+								string = string + ", " + aim;
+								$scope.eventDetails.aims[element.id] = string;
+							}
+						})
+					};//end
+
+					//mapping eventID to team names
+					if (element.relationships.teams.data.length > 0) {
+						element.relationships.teams.data.forEach( team => {
+							if ($scope.eventDetails.team[element.id] == undefined) {
+								$scope.eventDetails.team[element.id] = teamsIdToName[team.id];
+							} else {
+								var string = $scope.eventDetails.team[element.id];
+								string = string + ", " + teamsIdToName[team.id];
+								$scope.eventDetails.team[element.id] = string;
+							}
+						});
+					};//end
+
+					//mapping eventID to survey URI
+					if (element.relationships.survey.data != null) {
+						Restangular.one("surveys/" + element.relationships.survey.data.id)
+						.get()
+						.then(function(survey) {
+							$scope.eventDetails.survey[element.id] = survey.attributes.URI;
+						});
+					};//end
+
+					//mapping eventID to co-organizers
+					if (element.relationships.coorganizers.data != null) {
+						element.relationships.coorganizers.data.forEach( organization => {
+							if ($scope.eventDetails.coorganizers[element.id] == undefined) {
```

Comment:
Rather than do these if checks for all these mappings to add commas (if I've interpreted the point of these correctly), you can simply add each item to an array and after adding all the items do `array.join(', ');` This joins all the items of the array into a string, separated by `, ` (comma space)